core/chains/evm/logpoller/models.go

```diff
@@ -0,0 +1,34 @@
+package logpoller
+
+import (
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/lib/pq"
+
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// LogPollerBlock represents an unfinalized block
+// used for reorg detection when polling.
+type LogPollerBlock struct {
+	EvmChainId *utils.Big
+	BlockHash  common.Hash
+	// Note geth uses int64 internally https://github.com/ethereum/go-ethereum/blob/f66f1a16b3c480d3a43ac7e8a09ab3e362e96ae4/eth/filters/api.go#L340
+	BlockNumber int64
```

Comment:
yeah, I left it as an int64 for consistency with other chain related services. I'll see how it looks as a uint64

Reply:
hm on second thought seems better to align with the database type which is signed to avoid surprises. Adding a db constraint to make it positive seems nice though